dali/base/dasds.cpp

```diff
@@ -5695,49 +5699,113 @@ class CStoreHelper : implements IStoreHelper, public CInterface
-            OwnedIFileIO iFileIOTmpStore = createUniqueFile(location, TMPSAVENAME, NULL, tmpStoreName);
+            OwnedIFileIO iFileIOTmpStore = createUniqueFile(location, temporaryBinaryFileSaveName, NULL, tmpStoreName);
             OwnedIFile iFileTmpStore = createIFile(tmpStoreName);
             try
             {
+                // Serialize to memory buffer first to enable CRC calculation
+                Owned<IBufferedSerialOutputStream> out = createBufferedSerialOutputStream(memBuf);
+                root->serializeToStream(*out);
+                out->flush();
+                out.clear();
+
+                // Write to file through CRC stream
                 OwnedIFileIOStream fstream = createIOStream(iFileIOTmpStore);
                 Owned<ICrcIOStream> crcPipeStream = createCrcPipeStream(fstream);
-                Owned<IIOStream> ios = createBufferedIOStream(crcPipeStream);
-
-#ifdef _DEBUG
-                toXML(root, *ios);          // formatted (default)
-#else
-                toXML(root, *ios, 0, 0);
-#endif
-                ios->flush(); // ensure flushed outside of dtor (to ensure any exception thrown)
-                ios.clear();
-                fstream.clear();
-                crcPipeStream->flush(); // ensure flushed outside of dtor. NB: calls wrapped stream flush()
+                crcPipeStream->write(memBuf.length(), memBuf.toByteArray());
+                crcPipeStream->flush();
                 crc = crcPipeStream->queryCrc();
                 crcPipeStream.clear();
+                fstream.clear();
                 iFileIOTmpStore->close(); // ensure flushed outside of dtor
                 iFileIOTmpStore.clear();
             }
             catch (IException *e)
             {
-                DISLOG(e, "Exception(1) - Error saving store file");
+                DISLOG(e, "Exception - Error saving binary store file");
                 iFileIOTmpStore.clear();
                 iFileTmpStore->remove();
                 throw;
             }
+        }
+        catch (IException *e)
+        {
+            StringBuffer s("Exception - Error creating save files during saving binary store file");
+            DISLOG(e, s.str());
+            throw;
+        }
+
+        return crc;
+    }
+    virtual void saveStore(IPropertyTree *root, unsigned *_newEdition)
+    {
+        LOG(MCdebugProgress, "Saving store");
 
+        refreshStoreInfo();
+
+        unsigned edition = storeInfo.edition;
+        unsigned newEdition = nextEditionN(edition);
+        bool done = false;
+        try
+        {
+            StringBuffer tmpStoreName;
+            OwnedIFileIO iFileIOTmpStore = createUniqueFile(location, TMPSAVENAME, NULL, tmpStoreName);
+            OwnedIFile iFileTmpStore = createIFile(tmpStoreName);
+            unsigned crc = saveStoreToXMLFile(root, iFileIOTmpStore.get(), iFileTmpStore.get());
+            if (saveBinary)
+                saveStoreToBinaryFile(root);
```

Comment:
will also need to clearup the binary file and rename it..
so saveStoreToXMLFile and saveStoreToBinaryFile need to follow same pattern.

I suggest keeping the IFile's outside the outer try and removing the try/catch from within saveStoreToBinaryFile/saveStoreToXMLFile

Reply:
return (crc) value not being used.. it will need saving as an additional crc in writeStoreInfo